File: SortArray.java
import java.util.Arrays;

public class SortArray {
    /**
     * Sort an array of integers in ascending order in-place.
     *
     * @param nums Array of integers to sort
     */
    public static void sortArray(int[] nums) {
        Arrays.sort(nums);
    }

    public static void main(String[] args) {
        // Sample array
        int[] nums = {64, 34, 25, 12, 22, 11, 90};

        System.out.println("Original array: " + Arrays.toString(nums));
        sortArray(nums);
        System.out.println("Sorted array: " + Arrays.toString(nums));
    }
}

File: sort_array.go
package main

import (
	"fmt"
	"sort"
)

// sortArray sorts an array of integers in ascending order.
// It creates a copy of the input slice to avoid modifying the original.
func sortArray(nums []int) []int {
	// Create a copy of the slice
	sorted := make([]int, len(nums))
	copy(sorted, nums)

	// Sort the copy in ascending order
	sort.Ints(sorted)

	return sorted
}

func main() {
	// Sample slice
	originalSlice := []int{64, 34, 25, 12, 22, 11, 90}

	fmt.Println("Original slice:", originalSlice)
	sortedSlice := sortArray(originalSlice)
	fmt.Println("Sorted slice:", sortedSlice)
}
